### PR TITLE
Documentation update

### LINKED ISSUE #129: Don't store job data in /tmp
The default place to store the jobs is:

    SQLALCHEMY_DATABASE_URI = 'sqlite:////tmp/job_store.db'

and tmp is wiped on reboot (or even [daily](http://askubuntu.com/questions/20783/how-is-the-tmp-directory-cleaned-up/759048#759048))

This is no good, because it results in problems like this:
https://github.com/ckan/ckan/issues/3156

We should store it somewhere better by default.

### LINKED ISSUE #198: Replace sqlite with PostgreSQL
sqlite is not meant for concurrent access, and with the asynchronous nature of datapusher, jobs intermittently fail with database locking errors.

### LINKED ISSUE #200: Datapusher gets intermittently stuck when processing a large number of resources
**CKAN version:**
2.9

**Datapusher version:**
0.17

I'm using CKAN to create a human-readable version of several databases' system catalogs.

This entailed creating a crawler script that uses `ckanapi` to populate CKAN with hundreds of datasets, with corresponding CSVs.

However, Datapusher quickly gets stuck when the script processes these CSVs in a large batch, though it will be able to handle small batches without problems for the very same files.

At first, the problem was the use of sqlite for the job store, as sqlite was never meant for concurrent access, with intermittent database lock operational errors showing up in the datapusher.ERR file as datapusher updates the job store. (#198).

This was fixed by #199 .

Still, uwsgi was still running as a single process.  Even though the operational database lock errors were gone, datapusher was quickly overrun after processing a handful of CSVs.

### PR DESCRIPTION
**Pull request by amercader**
Fixes #198 
Fixes #200 
Fixes #129 

* Updated and simplified documentation, consolidated in the README
* Document all config options and defaults, add support for defining them via env vars
* Document concurrent setup with Postgres + uWSGI threads

### CODE REVIEW

**Comment by amercader** [comment]
@mbocevski @jqnatividad would love your input on this, particularly on the [High Availability Setup](https://github.com/ckan/datapusher/pull/207/files#diff-04c6e90faac2675aa89e2176d2eec7d8R119). @jqnatividad I think I captured all your proposed changes but let me know if you think something is missing.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+Add `datapusher` to the plugins in your CKAN configuration file
+(generally located at `/etc/ckan/default/production.ini` or `/etc/ckan/default/ckan.ini`):
+
+    ckan.plugins = <other plugins> datapusher
+
+In order to tell CKAN where this webservice is located, the following must be
+added to the `[app:main]` section of your CKAN configuration file :
+
+    ckan.datapusher.url = http://0.0.0.0:8800/
```
Comment (mbocevski) [inline README.md:160]: I think it's better the default URL to be http://127.0.0.1:8800 as is in the config files.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+Run the DataPusher::
+
+    python datapusher/main.py deployment/datapusher_settings.py
+
+By default DataPusher should be running at the following port:
+
+    http://localhost:8800/
+
+If you need to change the host or port, copy `deployment/datapusher_settings.py` to
+`deployment/datapusher_local_settings.py` and modify the file to suit your needs.
```
Comment (mbocevski) [inline README.md:66]: When changing the HOST and PORT of datapusher in settings, one must also change them in the uwsgi config as well https://github.com/ckan/datapusher/blob/master/deployment/datapusher-uwsgi.ini#L3.

**Comment by mbocevski** [comment]
@amercader Looks clean and I like moving the docs to the repo itself, I had 2 minor comments. Perhaps it's a good idea to include a section in the README about the uwsgi settings, cause we also are setting `uid` and `gid` to `www-data` there and depending on the system, this user/group combination might not exist. It's enough if we point to users to review the uwsgi config to make sure that it suits their system, prior to starting it.

**Comment by amercader** [comment]
Thanks @mbocevski I added all your suggestions

File: README.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+At this point you can run DataPusher with the following command:
+
+    /usr/lib/ckan/datapusher/bin/uwsgi -i /usr/lib/ckan/datapusher/src/datapusher/deployment/datapusher-uswgi.ini
+
+
+*Note*: If you are installing the DataPusher on a different location than the defaul
+one you need to adapt the relevant paths in the `datapusher-uwsgi.ini` to the ones you are using.
+
+
+### High Availability Setup
```
Comment (jqnatividad) [inline README.md:119]: Great! This section captures the essence of what I did in the two PRs.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+### DataPusher Configuration
+
+The DataPusher instance is configured in the `deployment/datapusher_settings.py` file.
+Here's a summary of the options available.
+
+| Name | Default | Description |
+| -- | -- | -- |
+| HOST | '0.0.0.0' | Web server host |
+| PORT | 8800 | Web server port |
+| SQLALCHEMY_DATABASE_URI | 'sqlite:////tmp/job_store.db' | SQLAlchemy Database URL. See note about database backend below. |
```
Comment (jqnatividad) [inline README.md:175]: We should change it from `/tmp` to `/var/lib/ckan` as @mbocevski suggested.

**Comment by abedkhooli** [comment]
Was this part of the 2.9.1 patch release? Can we move the /tmp/job_store.db to /var/lib/ckan in 2.9.1 or not yet (package install)?

**Comment by amercader** [comment]
@abedkhooli datapusher has a separate release cycle than CKAN, so it's not related to the patch releases. Unless I'm missing something you can change the db location to `/var/lib/ckan` or something else right now as long as the folder is writable by the user running datapusher

**Comment by abedkhooli** [comment]
@amercader thanks. I moved it to `/var/lib/ckan/jobstore/job_store.db` but getting `datapusher.jobs.HTTPError: <unprintable HTTPError object>` after it fetches the csv file, determines field types and `saving chunk 0 (last)`. The jobs table logs an error: internal server error which also shows in UI but no other clue what's going on.

**Pull request merged**